File: 10.C# MVC Frameworks - ASP.NET Core/05.Web API Introduction - Exercise/BookShop.Data/Models/Book.cs
namespace BookShop.Data.Models
{
    using System;
    using System.Collections.Generic;
    using System.ComponentModel.DataAnnotations;

    public class Book
    {
        public int Id { get; set; }

        [Required]
        [MinLength(DataConstants.BookTitleMinLenth)]
        [MaxLength(DataConstants.BookTitleMaxLenth)]
        public string Title { get; set; }

        [Required]
        public string Descrtioption { get; set; }

        [Range(0, double.MaxValue)]
        public decimal Price { get; set; }

        [Range(0,int.MaxValue)]
        public int Copies { get; set; }

        public int? Edition { get; set; }

        public int? AgeRestriction { get; set; }

        public DateTime ReleaseDate { get; set; }

        public int AuthorId { get; set; }

        public Author Author { get; set; }

        public List<CategoryBook> Categories { get; set; } = new List<CategoryBook>();
    }
}

File: 15.ExpressJS Fundamentals/04.View Engines - Exercise/02.BookLibrary/models/Book.js
const mongoose = require('mongoose')

let bookSheme = mongoose.Schema({
    title: { type: mongoose.Schema.Types.String, required: true },
    url: { type: mongoose.Schema.Types.String, required: true },
    releasedate: { type: mongoose.Schema.Types.String },
    author: { type: mongoose.Schema.Types.String }
})

let Book = mongoose.model('Book', bookSheme)

module.exports = Book